src/models/generative-model.ts

```diff
@@ -253,4 +255,54 @@ export class GenerativeModel {
       generativeModelRequestOptions,
     );
   }
+  
+  /**
+   * Connect to live streaming server. Returns an promise which resolves {@link LiveClient}
+   * A promise resolves when WebSocket connection are connected and client sent a config.
+   * @param [config={}] Live config.
+   * @param [connectionOptions={}] Live connectiing options. You can use a custom WebSocket API.
+   * 
+   * @example
+   * ```ts
+   * import { GoogleGenerativeAI } from "@google/generative-ai";
+   * 
+   * const model = new GoogleGenerativeAI("apiKey")
+   *   .getGenerativeModel({ model: "gemini-2.0-flash-exp" });
+   * 
+   * // Connect to live
+   * const live = await model.connectLive({
+   *   responseModalities: ["TEXT"]
+   * });
+   * 
+   * // Sending a text.
+   * live.send({
+   *   text: "Hi, what's your name?"
+   * });
+   * 
+   * for await (const event of live.listen()) {
+   *   // Handling a event.
+   *   console.log(event);
+   * }
+   * ```
+   */
+  connectLive(config: Omit<LiveConfig, 'model'> = {}, connectionOptions: LiveConnectionOptions = {}): Promise<LiveClient> {
+    return connect({
+      requestOptions: this._requestOptions,
+      apiKey: this.apiKey,
+      setup: {
+        ...config,
+        generationConfig: {
+          ...(config.generationConfig ?? {}),
+          ...this.generationConfig
+        },
+        model: this.model,
+        ...((config.tools || this.tools) ? {
+          tools: [
+            ...(config.tools ?? []),
+            ...(this.tools ?? [])
+          ]
+        } : {})
+      }
+    }, connectionOptions);
```

Comment:
```suggestion
    return connect(
      {
        requestOptions: this._requestOptions,
        apiKey: this.apiKey,
        setup: {
          ...config,
          generationConfig: {
            ...config.generationConfig,
            ...this.generationConfig,
          },
          model: this.model,
          ...(config.tools || this.tools
            ? {
                tools: [...config.tools, ...this.tools],
              }
            : {}),
        },
      },
      connectionOptions,
    );
```

Maybe this is smarter.